src/components/Filters/ToggleFilter.tsx

```diff
@@ -20,16 +20,19 @@ class ToggleFilter extends BaseFilter<boolean, ToggleFilterProps> implements Fil
     return (
       <Switch
         {...props}
-        selected={(value || false) === enabledValue}
+        // if the incoming `value` is undefined it means the switch has not been toggled
+        // - set it to false under normal behaviour
+        // - set it to true when enabledValue === false
+        selected={value === undefined ? false : enabledValue === false ? !value : value}
         label={this.label}
         labelStyle={inModal ? "filter" : "inline"}
-        onChange={(value) => {
-          // The switch being "off" might actually mean "true" to the backend.
-          // I.e. "Hide do not use" being "off" --> { includeDoNotUse: true }
-          const backendValue = !value ? !enabledValue : enabledValue;
-          // If the backend value is already the enabled value (i.e. the default), drop it
-          // I.e. `{ includeDoNotUse: false }` --> drop it, otherwise keep it.
-          setValue(backendValue === enabledValue ? undefined : backendValue);
+        onChange={(on) => {
+          // Basically, when the switch is off we return undefined
+          // to signify that no filtering should occur on this field
```

Comment:
This is a great comment; I was struggling to articulate what this behavior should be, but using "on and off" vs. "frontend vs. backend value" is much more clear. Thanks!